compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -22,9 +24,28 @@ use super::errors::{
 use super::{
     GenericArgsMode, ImplTraitContext, LoweringContext, ParamMode, ResolverAstLoweringExt,
 };
-use crate::errors::YieldInClosure;
+use crate::errors::{InvalidLegacyConstGenericArg, YieldInClosure};
 use crate::{fluent_generated, FnDeclKind, ImplTraitPosition};
 
+struct WillCreateDefIdsVisitor {}
+
+impl<'v> rustc_ast::visit::Visitor<'v> for WillCreateDefIdsVisitor {
+    type Result = ControlFlow<()>;
+
```

Comment:
This wont find any const arguments such as `foo::<{ 1 + 2 }>()` which will also be parented wrong. You can use `visit_anon_const` to find those then we can try running crater and see how widespread defining defs in legacy const generic args is

Reply:
Is this supposed to fail?
`std::arch::x86_64::_mm_blend_ps(loop {}, loop {}, foo::<3>());`

Reply:
Yeah, we desugar *every* const argument to an anon const right now

Reply:
(FWIW neither `3` nor `1 + 2` appears to ICE)

Reply:
Yeah anon consts don't seem to ICE but they will still have incorrect def parenting, I think defining *items* also doesn't ICE but will have incorrect def parenting. It's possible we'll wind up wanting to future compat warn on the stuff that doesnt ICE and error on the stuff that *does*. Though I think at some point we'll want to be having hir checks that def parenting lines up with how the hir is structured in which case all of these would be ICEing ^^